bin/jsii-release-golang

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+set -eu
+
+###
+#
+# Pushes a directory of golang modules to a GitHub repository.
+#
+# Usage: ./jsii-release-golang [DIR]
+#
+# DIR: directory where the golang modules are located (default is `dist/go`).
+#      Each subdirectory inside it must be a go module and contain a go.mod file.
+#
+# GITHUB_TOKEN       (required): Authentication token that allows to push and tag the repository.
+# VERSION            (required): Module version. Will be used as the repository tag. Required unless NO_TAG is set to a non-zero value.
+# GITHUB_REPO        (optional): GitHub repository to push to. Default is derived from the module name.
+# GIT_BRANCH         (optional): Branch to push to. Defaults to 'main'.
+# GIT_USER_NAME      (optional): Username to perform the commit with. Defaults to the current username (git config user.name)
+# GIT_USER_EMAIL     (optional): Email to perform the commit with. Defaults to the current email (git config user.email)
+# GIT_COMMIT_MESSAGE (optional): The commit message. Defaults to the last commit of the current repository.
+# DRYRUN             (optional): Set to "true" for a dry run
+###
+
+dir=${1:-dist/go}
+GIT_BRANCH=${GIT_BRANCH:-main}
+GIT_USER_NAME=${GIT_USER_NAME:-$(git config user.name)}
+GIT_USER_EMAIL=${GIT_USER_EMAIL:-$(git config user.email)}
+GIT_COMMIT_MESSAGE=${GIT_COMMIT_MESSAGE:-$(git log -1 --format=%s)}
```

Comment:
Don't assume we have a git repo because we normally release from a build artifact

Reply:
Removed this default